### ISSUE TITLE
V5.Users.CheckUserSubscriptionByChannelAsync and V5.Channels.CheckChannelSubscriptionByUserAsync throw BadResourceException when user is not subscribed

### ISSUE THREAD

**Issue by RRickkert** [TwitchLib/TwitchLib.Api#105]
Both the V5.Users.CheckUserSubscriptionByChannelAsync and V5.Channels.CheckChannelSubscriptionByUserAsync throw a BadResourceException when the provided user is not subscribed to the target channel.

The reason: Twitch returns a 404 with a message "[channel/user name] has no subscriptions to [channel/user name]" when the subscription cannot be found in both cases.

See the example responses if the User is Not Subscribed to the Channel: 
* https://dev.twitch.tv/docs/v5/reference/users/#check-user-subscription-by-channel
* https://dev.twitch.tv/docs/v5/reference/channels/#check-channel-subscription-by-user

The fact that TwitchLib throws a BadResourceException is unwanted, since the "twitch api resource can be found". It's just that there is no subscription.

This is very unclear, and there is no way of knowing what the actual problem is, since the message in the response cannot be obtained from the BadResourceException.

**Comment by RRickkert** [comment]
On the Twitch's side of the API, I would 100% agree that this is a correct way to implement it.
However, in the API, you would use this a lot to check IF a user is subscribed to a channel. Calling a method with correct parameters, credentials, etc. Should never throw an Exception. Exceptions should be what the name implies they are "exceptions". Something going wrong that wasn't entirely expected. Asking if a person is subscribed to a channel while he's not does not seem like an exceptional case to me. Along with that, the naming of the method does imply as well that it's checking if a user is subscribed. It's not implying it's retrieving subscription data and that it should only be called for users who you already know are subscribed.

Tldr: 404 return code is indeed ok on Twitch's side. Exceptions should not be thrown in normal use-case scenarios though. They should stay exceptions.

**Comment by Mahsaap** [comment]
I do tend to agree since people arnt clearly going to know an exception would be thrown and not know to catch such error. I think other things just return null if it's not present. There area few ways to go about it.

**Comment by RRickkert** [comment]
Even if the user would know that an exception will be thrown, it's quite bad to force the usage of try catch since the failure rate is so high (kinda like a 50/50 chance that a user is not subscribed).

Some more info on where my opinion is based upon: https://buildplease.com/pages/slow-exceptions/

**Comment by swiftyspiffy** [comment]
In line with the other v5 endpoint possible issues, I don't think we'll be fixing them because they're being turned off in a few months, and everyone should be using helix at this point.
